### ISSUE TITLE
SMTP-IP [doesn't work anymore]

### ISSUE THREAD

**Issue by vp-lab** [postalserver/postal#2444]
## Describe the bug

SMTP-IP feature isn't working anymore.

## To Reproduce

1. Go to your **Postal Interface**
2. Click **Credentials**
3. Click **Add another credential**
4. Create new **SMTP-IP** entry
5. Try to send mail without authentication

## Expected behaviour

If SMTP-IP it's set, I should be allow to sent mail without authentication.

## Screenshots

![2023-05-03_10-07](https://user-images.githubusercontent.com/18114871/235851764-e9b430ee-d3d7-4313-9a71-3b44c2734626.png)

## Environment details 

 - OS: AlmaLinux 8.7 (Stone Smilodon)
 - Version Postal: 2.1.4

## Additional information/context

SMTP-IP 

Logs from **smtp_server.log**
```
[9] [2023-05-03T06:33:26.206] DEBUG -- : [FISGOZ] .[35m   Connection opened from ::ffff:192.168.2.49.[0m
[9] [2023-05-03T06:33:26.206] DEBUG -- : [FISGOZ] .[35m   Client identified as ::ffff:192.168.2.49.[0m
[9] [2023-05-03T06:33:26.208] DEBUG -- : [FISGOZ] .[32m<= EHLO thunder.[0m
[9] [2023-05-03T06:33:26.210] DEBUG -- : [FISGOZ] .[34m=> 250-My capabilities are.[0m
[9] [2023-05-03T06:33:26.211] DEBUG -- : [FISGOZ] .[34m=> 250 AUTH CRAM-MD5 PLAIN LOGIN.[0m
[9] [2023-05-03T06:33:26.252] DEBUG -- : [FISGOZ] .[32m<= MAIL FROM:<>.[0m
[9] [2023-05-03T06:33:26.253] DEBUG -- : [FISGOZ] .[34m=> 250 OK.[0m
[9] [2023-05-03T06:33:26.254] DEBUG -- : [FISGOZ] .[32m<= RCPT TO:<intranet@postal.comtec.ro>.[0m
[9] [2023-05-03T06:33:26.259] DEBUG -- : [FISGOZ] .[34m=> 530 Authentication required.[0m
[9] [2023-05-03T06:33:26.261] DEBUG -- : [FISGOZ] .[32m<= RSET.[0m
[9] [2023-05-03T06:33:26.262] DEBUG -- : [FISGOZ] .[34m=> 250 OK.[0m
[9] [2023-05-03T06:33:26.263] DEBUG -- : [FISGOZ] .[32m<= QUIT.[0m
[9] [2023-05-03T06:33:26.264] DEBUG -- : [FISGOZ] .[34m=> 221 Closing Connection.[0m
[9] [2023-05-03T06:33:26.265] DEBUG -- : [FISGOZ] .[35m   Connection closed.[0m
```

**Comment by sinnbeck** [comment]
Just tested myself. Sent an email just fine using telnet

```
telnet gmail-smtp-in.l.google.com 25

HELO yourdomain.com

MAIL FROM:<user@yourdomain.com>

RCPT TO:<playingwithtelnet@gmail.com>

DATA
From: <user@yourdomain.com>
To: <playingwithtelnet@gmail.com>
Subject: Just a test email

The body of the mail goes here.          
.

QUIT

**Comment by vp-lab** [comment]
I followed your steps and I get ```530 Authentication required```

Logs
```
[9] [2023-05-04T06:47:48.231] DEBUG -- : [HZCPIA] .[35m   Connection opened from ::ffff:192.168.3.5.[0m
[9] [2023-05-04T06:47:48.232] DEBUG -- : [HZCPIA] .[35m   Client identified as ::ffff:192.168.3.5.[0m
[9] [2023-05-04T06:47:57.669] DEBUG -- : [HZCPIA] .[32m<= HELO postal.xxxxxxxxxx.ro.[0m
[9] [2023-05-04T06:47:57.673] DEBUG -- : [HZCPIA] .[34m=> 250 postal.xxxxxxxxxx.ro.[0m
[9] [2023-05-04T06:48:03.890] DEBUG -- : [HZCPIA] .[32m<= MAIL FROM:user@postal.xxxxxxxxx.ro.[0m
[9] [2023-05-04T06:48:03.891] DEBUG -- : [HZCPIA] .[34m=> 250 OK.[0m
[9] [2023-05-04T06:48:09.548] DEBUG -- : [HZCPIA] .[32m<= RCPT TO:vp@xxxxxxxx.ro.[0m
[9] [2023-05-04T06:48:09.553] DEBUG -- : [HZCPIA] .[34m=> 530 Authentication required.[0m
```
![2023-05-04_09-50](https://user-images.githubusercontent.com/18114871/236131636-3ac22f41-6f47-4c58-a8bf-8bbf60ed14f1.png)
![2023-05-04_09-54](https://user-images.githubusercontent.com/18114871/236131729-2b3a6d70-6a6f-425a-9d66-5ef525274e8b.png)

**Comment by sinnbeck** [comment]
Does the domain you are sending from in telnet match the domain in the server in postal?

**Comment by vp-lab** [comment]
Yes, it's the same domain.

**Comment by sinnbeck** [comment]
Ok. Maybe someone else can recreate it as I cannot. Only difference for me is that I am not doing it on a local network (as its running on an external server). I just setup SMTP-IP for my own external ip and it worked perfectly.

**Comment by kaiserto** [comment]
I have the same Problem and probably found an workaround. When you enter the in from ::ffff:192.168.3.5 it will work.

**Comment by ArriMa** [comment]
Hello, we have the same issue with SMTP-IP.
adding "::ffff:" before the IP or subnet start working again. thanks @kaiserto.

From the application side the error is "530 Authentication required" even if is configured with SMTP-IP

**Comment by vp-lab** [comment]
I can confirm, it's working again with ```::ffff:``` before the IP.

**Comment by catphish** [comment]
Confirmed. This should be fixed by 8b525d0381a9e0113af808b9ec2eb47bf78ec60b
